src/allmydata/crypto/rsa.py

```diff
@@ -0,0 +1,157 @@
+from cryptography.exceptions import InvalidSignature
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.asymmetric import rsa, padding
+from cryptography.hazmat.primitives.serialization import load_der_private_key, load_der_public_key, \
+    Encoding, PrivateFormat, PublicFormat, NoEncryption
+
+from allmydata.crypto.error import BadSignature
+
+
+# This is the value that was used by `pycryptopp`, and we must continue to use it for
+# both backwards compatibility and interoperability.
+#
+# The docs for `cryptography` suggest to use the constant defined at
+# `cryptography.hazmat.primitives.asymmetric.padding.PSS.MAX_LENGTH`, but this causes old
+# signatures to fail to validate.
```

Comment:
FWIW, this salt length appears to follow the convention of being equal to the length of the hash function output.

It is dramatically smaller than the value cryptography would use, which appears to be 222:

```
>>> priv, pub = allmydata.crypto.rsa.create_signing_keypair(2048)
>>> cryptography.hazmat.primitives.asymmetric.padding.calculate_max_pss_salt_length(priv, allmydata.crypto.rsa.hashes.SHA256())
222
```

but as pointed out changing it here is difficult because PSS has external parameters (signatures don't completely describe themselves).  If we want to change this or any other PSS parameters we would probably need to invent a "priv-v1-..." key which we might want to do - but that's entirely separate from this effort.

Reply:
Yeah, I don't want to change anything like that in this PR .. does it need more comments, or ..?